src/controllers/item.js

```diff
@@ -11,13 +18,41 @@ module.exports = {
   getFilteredItems: (req, res) => {
     res.send('Filtered items');
   },
-  updateItem: (req, res) => {
-    res.send('Item with the specified ID has been updated');
+  updateItem: async (req, res) => {
+    const { id } = req.params;
+    try {
+      const updatedItem = await ItemModel.findByIdAndUpdate(
+        id,
+        { $set: req.body },
+        { new: true }
+      );
+      if (!updatedItem) {
+        res
+          .status(422)
+          .json({ message: 'The item with the specified ID was not found.' });
+      } else {
+        res.json(updatedItem);
+      }
```

Comment:
if you throw an error in the if statement, there will be no need to use else anymore, you just need to write `res.json(updatedItem);` because when an error is thrown, it will jump directly to `catch` statement